### ISSUE TITLE
real player should be limited to inviting a maximum bots and the master of the bots should be set as the player who invited them

### ISSUE THREAD

**Issue by zbhcn** [liyunfan1223/mod-playerbots#335]
Each real player should be limited to inviting a maximum of 4 or N bots into their group. In a raid group, the master of the bots should be set as the player who invited them or their assistant, rather than always being the team leader.

**Comment by zbhcn** [comment]
Or there should be a configuration parameter switch that limits the maximum number of invited bots, as it is necessary when there are multiple players playing on the server

**Comment by noisiver** [comment]
It shouldn't be a limitation by default because that makes no sense. That way you couldn't do 10, 15, 20, 25 or 40 man raids with bots on your own.
Adding a config option to limit the number of bots each player can be a master of makes more sense. It should be unlimited by default though.

**Comment by zbhcn** [comment]
only raid leader can contrl bots when there are multiple players playing in the raid group ,it's not good.Usually, only one player watches 24 bots fight against the boss, which is even more meaningless.Players have nothing to do, boss who can win watch bot battles, while boss who cannot win still cannot change the outcome

**Comment by zbhcn** [comment]
> only raid leader can contrl bots when there are multiple players playing in the raid group ,it's not good.Usually, only one player watches 24 bots fight against the boss, which is even more meaningless.Players have nothing to do, boss who can win watch bot battles, while boss who cannot win still cannot change the outcome



> It shouldn't be a limitation by default because that makes no sense. That way you couldn't do 10, 15, 20, 25 or 40 man raids with bots on your own. Adding a config option to limit the number of bots each player can be a master of makes more sense. It should be unlimited by default though.

**Comment by Dreathean** [comment]
1) While I wouldn't use it for my purposes, I definitely see how a configurable option for larger servers would be extremely useful

2) Regarding that last comment, you're sure that if two (or more players) who each are the master of multiple bots, that in a raid only the raid leader now controls ALL bots? I haven't tested that yet, but I am pretty sure in a party setting, the party leader did not control all bots with multiple masters. Will look more into it when I get a chance though.

**Comment by zbhcn** [comment]
> 1. While I wouldn't use it for my purposes, I definitely see how a configurable option for larger servers would be extremely useful
> 2. Regarding that last comment, you're sure that if two (or more players) who each are the master of multiple bots, that in a raid only the raid leader now controls ALL bots? I haven't tested that yet, but I am pretty sure in a party setting, the party leader did not control all bots with multiple masters. Will look more into it when I get a chance though.

yes,you should try a situation where there are multiple real players in the team, and only the raid group leader can control the bots

**Comment by Dreathean** [comment]
Yup I will try it out when I get to play with my friend but it will be a while, if that's the case then that definitely needs to get fixed.
